### PR TITLE
Notebooks: Change Default Max Table Rows Returned in Notebooks to 5000

### PR DESCRIPTION
**Pull request by chlafreniere**
For notebooks, change default max table rows returned to 5000.

Also added a user-configurable setting to change this if users so desire.

### CODE REVIEW

**Comment by anthonydresser** [comment]
Is there really a performance impact on the amount of rows there is? In the query results its always been the amount of grids, not really the amount of rows per grid.

**Comment by chlafreniere** [comment]
Totally get what you're saying; we do actually have perf issues at the moment with large result sets, since we need to get the data back, convert it into 2 different MIME types (to ensure compatibility with other viewers), and finally load it into a table. A quick test with 30k rows returned showed that it took ~12 seconds to show the table, and even loading the notebook (since we save all of the data inside) took a few seconds with only that 1 table. If you have 10 (or 100) tables in your notebook, we still have work to do in order to provide for a more performant experience.

**Comment by anthonydresser** [comment]
Do you have an example notebook document I can use to look into this? I'd be interested in seeing the issues.

**Comment by chlafreniere** [comment]
Sent you a mail with the repro (since it includes sensitive info like server names, etc)

**Comment by kevcunnane** [comment]
Yeah this is mostly a "how big a file do we want to save, and can safely reopen later?" issue. We're going with a default value for now and long-term might want to support "show all the data on initial query, truncate on save"version of the world.

File: src/sql/workbench/services/notebook/common/sqlSessionManager.ts [hunk 1]
```diff
@@ -270,9 +274,17 @@ export class SQLFuture extends Disposable implements FutureInternal {
 	private ioHandler: nb.MessageHandler<nb.IIOPubMessage>;
 	private doneHandler: nb.MessageHandler<nb.IShellMessage>;
 	private doneDeferred = new Deferred<nb.IShellMessage>();
+	private configuredMaxRows: number = MAX_ROWS;
 
-	constructor(private _queryRunner: QueryRunner, private _executionCount: number | undefined) {
+	constructor(private _queryRunner: QueryRunner, private _executionCount: number | undefined, private configurationService: IConfigurationService) {
 		super();
+		let config = configurationService.getValue(NotebookConfigSectionName);
```
Comment (kevcunnane) [inline src/sql/workbench/services/notebook/common/sqlSessionManager.ts]: Minor: might want to update this value before running a query. Users expect changed to happen immediately if possible and not require reopening an editor.

Comment (chlafreniere) [reply src/sql/workbench/services/notebook/common/sqlSessionManager.ts]: Wait, doesn't this happen already? I had a notebook open, changed/saved the setting, went back to the notebook, and the notebook returned the max rows that I was expecting.

Comment (kevcunnane) [reply src/sql/workbench/services/notebook/common/sqlSessionManager.ts]: If it's in the constructor it'll only update on creating a new session, correct?

Comment (chlafreniere) [reply src/sql/workbench/services/notebook/common/sqlSessionManager.ts]: it's whenever we create a new future, which currently is on any new query. Should we not be creating a new future every time?

**Pull request merged**